src/main/java/org/alliancegenome/curation_api/services/validation/dto/Gff3DtoValidator.java

```diff
@@ -0,0 +1,177 @@
+@RequestScoped
+public class Gff3DtoValidator {
+
+	@Inject ExonDAO exonDAO;
+	@Inject TranscriptDAO transcriptDAO;
+	@Inject CodingSequenceDAO codingSequenceDAO;
+	@Inject DataProviderService dataProviderService;
+	@Inject NcbiTaxonTermService ncbiTaxonTermService;
+	@Inject SoTermDAO soTermDAO;
+	
+	@Transactional
+	public Exon validateExonEntry(Gff3DTO dto, BackendBulkDataProvider dataProvider) throws ObjectValidationException {
+
+		Exon exon = null;
+		
+		Map<String, String> attributes = getAttributes(dto, dataProvider);
+		String uniqueId = Gff3UniqueIdHelper.getExonOrCodingSequenceUniqueId(dto, attributes, dataProvider);
+		SearchResponse<Exon> searchResponse = exonDAO.findByField("uniqueId", uniqueId);
+		if (searchResponse != null && searchResponse.getSingleResult() != null) {
+			exon = searchResponse.getSingleResult();
+		} else {
+			exon = new Exon();
+			exon.setUniqueId(uniqueId);
+		}
+		
+		if (attributes.containsKey("Name")) {
+			exon.setName(attributes.get("Name"));
+		}
+		
+		ObjectResponse<Exon> exonResponse = validateGffEntity(exon, dto, attributes, dataProvider);
+	
+		if (exonResponse.hasErrors()) {
+			throw new ObjectValidationException(dto, exonResponse.errorMessagesString());
+		}
+		
+		return exonDAO.persist(exonResponse.getEntity());
+	}
+	
+	@Transactional
+	public CodingSequence validateCdsEntry(Gff3DTO dto, BackendBulkDataProvider dataProvider) throws ObjectValidationException {
+
+		CodingSequence cds = null;
+		
+		Map<String, String> attributes = getAttributes(dto, dataProvider);
+		String uniqueId = Gff3UniqueIdHelper.getExonOrCodingSequenceUniqueId(dto, attributes, dataProvider);
+		SearchResponse<CodingSequence> searchResponse = codingSequenceDAO.findByField("uniqueId", uniqueId);
+		if (searchResponse != null && searchResponse.getSingleResult() != null) {
+			cds = searchResponse.getSingleResult();
+		} else {
+			cds = new CodingSequence();
+			cds.setUniqueId(uniqueId);
+		}
+		
+		if (attributes.containsKey("Name")) {
+			cds.setName(attributes.get("Name"));
+		}
+		
+		ObjectResponse<CodingSequence> cdsResponse = validateGffEntity(cds, dto, attributes, dataProvider);
+	
+		if (cdsResponse.hasErrors()) {
+			throw new ObjectValidationException(dto, cdsResponse.errorMessagesString());
+		}
+		
+		return codingSequenceDAO.persist(cdsResponse.getEntity());
+	}
+	
+	@Transactional
+	public Transcript validateTranscriptEntry(Gff3DTO dto, BackendBulkDataProvider dataProvider) throws ObjectValidationException {
+
+		Transcript transcript = null;
+		
+		Map<String, String> attributes = getAttributes(dto, dataProvider);
+		if (attributes.containsKey("ID")) {
+			SearchResponse<Transcript> searchResponse = transcriptDAO.findByField("modInternalId", attributes.get("ID"));
+			if (searchResponse != null && searchResponse.getSingleResult() != null) {
+				transcript = searchResponse.getSingleResult();
+			} else {
+				transcript = new Transcript();
+				transcript.setModInternalId(attributes.get("ID"));
+			}	
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line must not have trailing whitespace